x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/step_define_package_policy.tsx

```diff
@@ -209,7 +209,17 @@ export const StepDefinePackagePolicy: React.FunctionComponent<{
           {/* Required vars */}
           {requiredVars.map((varDef) => {
             const { name: varName, type: varType, default: defaultValue } = varDef;
-            const value = packagePolicy.vars?.[varName]?.value ?? defaultValue;
+
+            // Set up the `vars` object and assign the default value if it doesn't exist
+            if (!packagePolicy.vars) {
```

Comment:
it seems odd to me to reassign vars in a component render, can we do this in a `useEffect` calling `onChange` ?